proto/spire/api/server/federationmanagement/v1/federationmanagement.proto

```diff
@@ -0,0 +1,154 @@
+syntax = "proto3";
+package spire.api.server.federationmanagement.v1;
```

Comment:
I'm wondering about the API name ...

If we are going to stick to the "federation" term, it's hard for me to come up with something better than this ... but maybe there are other ways to look at it? For example, maybe trust domain API makes sense? Or bundle endpoint API?

I think part of our difficulty here is that with the other APIs, we are resource-centric ... but, for federation, we never really came up with a good name for this resource that is being managed. In the config, we used `federate_with` ... `federation relationship` could work, but I wonder what else we could call them.

Reply:
>I think part of our difficulty here is that with the other APIs, we are resource-centric

Yes, that's a really good point. I've been thinking about this and Trust Domain API seems to be the most appropriate name IMO. I prefer Trust Domain API instead of Bundle Endpoint API because 1) We are really managing the relationships between trust domains, not the bundle endpoints themselves. 2) It's out of the scope of the API to manage the bundle endpoint server. The API deals with bundle endpoints as a way to manage relationships between trust domains.